docs/markdown/06-iterator_generator/01-Iterator.md

```diff
@@ -54,9 +51,69 @@ for (elem of myArray) {
 
 ##--##
 
-![h-400 center](./assets/images/Iterable_01_array.png) <!-- .element: class="fragment" data-fragment-index="1" -->
+<!-- .slide -->
+
+<br />
+<br />
+
+![h-400 center](./assets/images/Iterable_01_array.png) <!-- .element: class="fragment" -->
 
 Notes:
-à utiliser sur des itérables (not only array ;) )
 
 un iterator renvoie 2 choses, si l'itérateur est fini et la valeur suivante
+{value: 'foo', done: false}
+
+##==##
+
+<!-- .slide: class="with-code" -->
+
+# Itérable : Comment les parcourir
+
+<br />
+
+## Le spread operator
+
+```javascript
+let mySet = new Set([1, 2, 3]);
+let values = [...mySet]; // [1, 2, 3]
+```
+<!-- .element: class="fragment" -->
+
+##==##
+
+<!-- .slide: class="with-code" -->
+
+# Itérable : Comment les parcourir
+
+<br />
+
+## La boucle while
+
+```javascript
+let iterableObject = {
```

Comment:
est ce qu'on peut pas avoir un exemple plus simple ? ou alors faut expliquer en premier lieu le fonctionnement d'un iterator avec symbol.iterator.... Car au final on parle d'iterator, avec for of, ça semble tout simple, avec while c'est pas forcément compliqué c'est juste que l'implémentation autour de la transformation de l'objet en iterator n'est pas simple, ce qui nous fait perdre la compréhension du while...

Donc j'aurais fait un slide avant avec le code de l'iterator en mode, object javascript basique, et objet javascript iterable.

Reply:
dans ce cas je devrais mettre une note sur le for of pour expliquer que c'est du sucre syntaxique pour masquer ce qui serait sinon proche de cet exemple avec while.
```
let myArray = [1, 2, 'foo', 3];
let iterator = myArray[Symbol.iterator](), next = iterator.next();
while(!next.done) {
  console.log(next.value);
  next = iterator.next();
}
```
De cette manière j'aurais déjà montré le while en version simple, ce qui permettra de se focaliser ici sur mon objectif premier qui était de montrer que ça marche avec tout type d'iterable, ici un iterable custom.